### ISSUE TITLE
Warn about undefined external symbols just like for functions?

### ISSUE THREAD

**Issue by fbrausse** [esbmc/esbmc#1424]
> I've double-checked an indeed forgot to include some of MUSL's constants in the runs.

_Originally posted by @fbrausse in https://github.com/esbmc/esbmc/issues/818#issuecomment-1780626386_

This was about a symbol defined `extern` and thus silently treated as nondet by ESBMC. When functions are being called that have no body available a message is printed. In contrast for accesses to such external symbols no information is given in the log.

Is this intentional? It would have been easier for me to run the correct command line if this information had been there.

**Comment by rafaelsamenezes** [comment]
> Is this intentional?

I guess, we are just no explicit about it. I see no problem in advise the user that some functions are not defined.

**Comment by fbrausse** [comment]
> I guess, we are just no explicit about it.

Indeed, that's what this question is about: can we be explicit in the log about usage of external symbols? I guess the linker might be able to give an overapproximation here.
